crates/runtime/src/dialogue.rs

```diff
@@ -22,146 +19,67 @@ mod shared_state;
-        self
-    }
-
-    /// The [`OptionsHandler`] that is called when a set of options are ready to be shown to the user.
-    ///
-    /// The Options Handler delivers a [`Vec`] of [`DialogueOption`] to the game.
-    /// Before [`Dialogue::continue_`] can be called to resume execution,
-    /// [`Dialogue::set_selected_option`] must be called to indicate which
-    /// [`DialogueOption`] was selected by the user. If [`Dialogue::set_selected_option`] is not called, a panic occurs.
-    pub fn set_options_handler(
-        &mut self,
-        options_handler: impl FnMut(Vec<DialogueOption>, &mut HandlerSafeDialogue)
-            + 'static
-            + Send
-            + Sync,
-    ) -> &mut Self {
-        self.vm.options_handler = Box::new(options_handler);
-        self
-    }
-
-    /// The [`CommandHandler`] that is called when a command is to be delivered to the game.
-    pub fn set_command_handler(
-        &mut self,
-        command_handler: impl FnMut(Command, &mut HandlerSafeDialogue) + 'static + Send + Sync,
-    ) -> &mut Self {
-        self.vm.command_handler = Box::new(command_handler);
-        self
-    }
-
-    /// The [`NodeCompleteHandler`] that is called when a node is complete.
-    pub fn set_node_complete_handler(
-        &mut self,
-        node_complete_handler: impl FnMut(String, &mut HandlerSafeDialogue) + 'static + Send + Sync,
-    ) -> &mut Self {
-        self.vm.node_complete_handler = Box::new(node_complete_handler);
-        self
-    }
-
-    /// The [`NodeStartHandler`] that is called when a node is started.
-    pub fn set_node_start_handler(
-        &mut self,
-        node_start_handler: impl FnMut(String, &mut HandlerSafeDialogue) + 'static + Send + Sync,
-    ) -> &mut Self {
-        self.vm.node_start_handler = Some(Box::new(node_start_handler));
-        self
-    }
-
-    /// The [`DialogueCompleteHandler`] that is called when the Dialogue reaches its end.
-    pub fn set_dialogue_complete_handler(
-        &mut self,
-        dialogue_complete_handler: impl FnMut(&mut HandlerSafeDialogue) + 'static + Send + Sync,
-    ) -> &mut Self {
-        self.vm.dialogue_complete_handler = Some(Box::new(dialogue_complete_handler));
+    pub fn with_variable_storage(
+        mut self,
+        variable_storage: impl VariableStorage + 'static + Send + Sync,
+    ) -> Self {
+        self.set_variable_storage(variable_storage);
         self
     }
 
-    /// The [`PrepareForLinesHandler`] that is called when the dialogue anticipates delivering some lines.
-    pub fn set_prepare_for_lines_handler(
-        &mut self,
-        prepare_for_lines_handler: impl FnMut(Vec<LineId>, &mut HandlerSafeDialogue)
-            + 'static
-            + Send
-            + Sync,
-    ) -> &mut Self {
-        self.vm.prepare_for_lines_handler = Some(Box::new(prepare_for_lines_handler));
+    pub fn with_language_code(mut self, language_code: impl Into<String>) -> Self {
```

Comment:
Why do some of these functions take `&mut self` and some `mut self`?

Reply:
[`ded3f30` (#103)](https://github.com/yarn-slinger/yarn-slinger/pull/103/commits/ded3f3095282dff5a6d9144ec5d69668e134dfd1)